plugin.py

```diff
@@ -1,99 +1,97 @@
+    server.setup()
 
 
-def is_node_installed():
-    return shutil.which('node') is not None
+def plugin_unloaded():
+    server.cleanup()
 
 
-def logAndShowMessage(msg, additional_logs=None):
-    print(msg, '\n', additional_logs) if additional_logs else print(msg)
-    sublime.active_window().status_message(msg)
+def is_node_installed():
+    return shutil.which('node') is not None
 
 
 class LspJSONPlugin(LanguageHandler):
     @property
     def name(self) -> str:
-        return 'lsp-json'
+        return PACKAGE_NAME.lower()
 
     @property
     def config(self) -> ClientConfig:
-        settings = sublime.load_settings("LSP-json.sublime-settings")
-        client_configuration = settings.get('client')
+        # Calling setup() also here as this might run before `plugin_loaded`.
+        # Will be a no-op if already ran.
+        # See https://github.com/sublimelsp/LSP/issues/899
+        server.setup()
+
+        configuration = self.migrate_and_read_configuration()
+
         default_configuration = {
-            "command": [
-                'node',
-                server_path,
-                '--stdio'
-            ],
-            "languages": [
+            'enabled': True,
+            'command': ['node', server.binary_path, '--stdio'],
+            'languages': [
                 {
-                    "languageId": "json",
-                    "scopes": ["source.json"],
-                    "syntaxes": [
-                        "Packages/JavaScript/JSON.sublime-syntax",
-                        "Packages/JSON/JSON.sublime-syntax"
+                    'languageId': 'json',
+                    'scopes': ['source.json'],
+                    'syntaxes': [
+                        'Packages/JavaScript/JSON.sublime-syntax',
+                        'Packages/JSON/JSON.sublime-syntax'
                     ]
                 }
-            ]
+            ],
+            'settings': {
+                'json': {
+                    'format': {
+                        'enable': True
+                    }
+                }
+            }
         }
-        default_configuration.update(client_configuration)
+
+        default_configuration.update(configuration)
+
+        if 'json' not in default_configuration['settings']:
+            default_configuration['settings']['json'] = {}
         default_configuration['settings']['json']['schemas'] = schemas
-        return read_client_config('lsp-json', default_configuration)
+
+        return read_client_config(self.name, default_configuration)
+
+    def migrate_and_read_configuration(self) -> None:
+        settings = {}
+        loaded_settings = sublime.load_settings(SETTINGS_FILENAME)
+
+        if loaded_settings:
+            if loaded_settings.has('client'):
+                client = loaded_settings.get('client')
+                loaded_settings.erase('client')
+                # Migrate old keys
+                for key in ['languages', 'initializationOptions', 'settings']:
```

Comment:
```
                # Migrate old keys
                for key in client:
                    loaded_settings.set(key, client[key])
                sublime.save_settings(SETTINGS_FILENAME)
```
This here can be just a bit simplified... a bit :)

Reply:
Thanks. Applied.
I also noticed there was a bug with not reading `languages` key in the code below.